frontend/src/pages/UserEditPage.jsx

```diff
@@ -1,14 +1,361 @@
+            <p className='text-error text-sm mt-1'>
+              {formErrors.repeatNewPassword}
+            </p>
+          )}
+        </div>
+
+        <div className='mb-4'>
+          <label className='block text-lg font-medium text-gray-700 mb-2'>
+            Admin Status:
+          </label>
+          <div className='flex items-center space-x-4'>
+            <label className='inline-flex items-center'>
+              <input
+                type='radio'
+                name='is_admin'
+                value='true'
+                checked={formData.is_admin === true}
+                onChange={handleChange}
+                className='radio radio-primary'
+                disabled={isUpdating}
+              />
+              <span className='ml-2 text-gray-700'>Yes</span>
+            </label>
+            <label className='inline-flex items-center'>
+              <input
+                type='radio'
+                name='is_admin'
+                value='false'
+                checked={formData.is_admin === false}
+                onChange={handleChange}
+                className='radio radio-primary'
+                disabled={isUpdating}
+              />
+              <span className='ml-2 text-gray-700'>No</span>
+            </label>
+          </div>
+        </div>
+
+        {showInlineConfirm && (
+          <div
+            className='bg-yellow-100 border-l-4 border-yellow-500 text-yellow-700 p-4 mb-4'
+            role='alert'
+          >
+            <p className='font-bold'>Confirm Update</p>
+            <p>Are you sure you want to save these changes?</p>
+            {updateError && (
+              <p className='text-error mt-2'>Error: {updateError.message}</p>
+            )}
+            <div className='flex justify-end gap-4 mt-4'>
+              <button
+                type='button'
+                onClick={cancelUpdate}
+                className='animated-button !text-base !px-5 !py-2 !bg-gray-500 hover:!bg-gray-600'
+                disabled={isUpdating}
+              >
+                Cancel
+              </button>
+              <button
+                type='button'
+                onClick={confirmUpdate}
+                className='animated-button !text-base !px-5 !py-2'
+                disabled={isUpdating}
+              >
+                {isUpdating ? 'Confirming...' : 'Confirm Save'}
+              </button>
+            </div>
+          </div>
+        )}
+
+        <div className='flex justify-center gap-4 mt-6'>
+          <button
+            type='button'
+            onClick={() => navigate(`/users/${id}`)}
+            className='animated-button !text-base !px-5 !py-2 !bg-gray-500 hover:!bg-gray-600'
+            disabled={isUpdating}
+          >
+            Back to User Profile
+          </button>
+          <button
+            type='submit'
+            className='animated-button !text-base !px-5 !py-2'
+            disabled={isUpdating || showInlineConfirm}
```

Comment:
Ideally save would be disabled also when no changes to the initial data was made. This could be left for separate PR. What do you think?